src/browser/atom-application.coffee

```diff
@@ -411,6 +415,41 @@ class AtomApplication
         console.log("Killing process #{pid} failed: #{error.code ? error.message}")
     delete @pidsToOpenWindows[pid]
 
+  saveState: ->
+    states = []
+    for window in @windows
+      if loadSettings = window.getLoadSettings()
+        states.push(_.pick(loadSettings,
+          'initialPaths'
+          'isSpec'
+          'devMode'
+          'safeMode'
+          'apiPreviewMode'
+        ))
+    fs.writeFileSync(@getStatePath(), JSON.stringify(states))
+
+  loadState: ->
+    try
+      stateString = fs.readFileSync(@getStatePath(), 'utf8')
+    catch error
+      if error.code is 'ENOENT'
+        return false
+      else
+        throw error
+
+    states = JSON.parse(fs.readFileSync(@getStatePath()))
```

Comment:
Yeah, thanks. I think I'm going to :fire: this and extract a coupe of helper functions for working with the `storage` directory here and in `atom.coffee`.